pkg/containers/containers.go

```diff
@@ -88,63 +90,64 @@ func (c *Containers) GetCgroupV1HID() int {
 
 // Populate populates Containers struct by reading mounted proc and cgroups fs.
 func (c *Containers) Populate() error {
-	err := c.procMountsCgroups()
-	if err != nil {
-		return err
+	if c.cgroupMP == "" {
+		err := c.findCgroupMounts()
+		if err != nil {
+			return err
+		}
 	}
 
 	return c.populate()
 }
 
 // getCgroupV1SSID finds cgroup v1 hierarchy ID.
-func initCgroupV1() error {
-	cgroupsFile := "/proc/cgroups"
-	file, err := os.Open(cgroupsFile)
+func (c *Containers) initCgroupV1() error {
+	c.cgroupV1 = true
+	hierarchyId, err := getCgroupV1HierarchyId()
 	if err != nil {
-		return fmt.Errorf("could not open cgroups file %s: %w", cgroupsFile, err)
+		return err
 	}
-	defer file.Close()
+	cgroupV1HierarchyID = hierarchyId
 
-	scanner := bufio.NewScanner(file)
-	for i := 1; scanner.Scan(); i++ {
-		sline := strings.Fields(scanner.Text())
-		if len(sline) < 2 || sline[0] != cgroupV1Controller {
-			continue
+	inContainer, err := runsOnContainerV1()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "containers: failed to detect if running on a cgroupv1 container: %s", err)
+	}
+
+	if inContainer {
```

Comment:
if `err != nil` above, the value of this variable is invalid, isn't it?

Reply:
By convention the bool returned in such a case would be false, which is also the case here. I can add a return there anyway if you want.

Reply:
Added the early return.